From 86216a5ee3458e90fd92321532086c3c6b8ea041 Mon Sep 17 00:00:00 2001
From: T S Rameshkumar <rameshkumar.t@phytecembedded.in>
Date: Tue, 13 Oct 2026 17:52:35 +0530
Subject: [PATCH 5/7] board: phytec: phycore_imx8mm: Add industrial auto-boot
 command for Android 15

Configure U-Boot default environment and header files for phyBOARD-Polis
to automatically load and boot Android 15 kernel and device tree from MMC
storage on power-on. Clean up legacy Yocto/RAUC variables from phycore_imx8mm.env.

Signed-off-by: T S Rameshkumar <rameshkumar.t@phytecembedded.in>
---
 .../phytec/phycore_imx8mm/phycore_imx8mm.env  | 136 +++---------------
 configs/phycore-imx8mm_android_defconfig      |   2 +-
 configs/phycore-imx8mm_android_uuu_defconfig  |   2 +-
 include/configs/phycore_imx8mm_android.h      |   8 ++
 4 files changed, 26 insertions(+), 122 deletions(-)

diff --git a/board/phytec/phycore_imx8mm/phycore_imx8mm.env b/board/phytec/phycore_imx8mm/phycore_imx8mm.env
index f95c91fca72..dd18d18ddf1 100644
--- a/board/phytec/phycore_imx8mm/phycore_imx8mm.env
+++ b/board/phytec/phycore_imx8mm/phycore_imx8mm.env
@@ -1,123 +1,19 @@
-bootcmd=
-	run spiprobe;
-	if test ${dolegacyboot} = 1; then
-		run legacyboot;
-	fi;
-	bootflow scan -lb;
-	if test ${doefiboot} = 1; then
-		setenv bootmeths "efi_mgr efi";
-		setenv boot_targets "mmc2 mmc1 usb";
-		bootflow scan -l;
-		bootflow scan -lb;
-	fi;
-legacyboot=
-	mmc dev ${mmcdev};
-	if mmc rescan; then
-		if test ${doraucboot} = 1; then
-			run raucinit;
-		fi;
-		if run loadimage; then
-			run mmcboot;
-		else
-			if test ${doraucboot} = 1 || test ${dofitboot} = 1; then
-				reset;
-			fi;
-			run netboot;
-		fi;
-	fi;
-boot_targets=mmc2 mmc1 ethernet
-boot_script_dhcp=boot.scr.uimg
-bootmeths=script
-bootenv_addr_r=0x49100000
+bootdelay=2
 console=ttymxc2,115200
-doefiboot=1
-dolegacyboot=0
 emmc_dev=2
 sd_dev=1
-fdt_addr_r=0x48000000
-fdtfile=CONFIG_DEFAULT_FDT_FILE
-fit_fdtconf=conf-CONFIG_DEFAULT_DEVICE_TREE.dtb
-fdtoverlay_addr_r=0x49000000
-image=Image
-ipaddr=192.168.3.11
-serverip=192.168.3.10
-netmask=255.225.255.0
-ip_dyn=no
-kernel_addr_r=0x5A080000
-kernel_comp_addr_r=0x60000000
-kernel_comp_size=0x2000000
-loadimage=fatload mmc ${mmcdev}:${mmcpart} ${loadaddr} ${image}
-loadfdt=fatload mmc ${mmcdev}:${mmcpart} ${fdt_addr_r} ${fdtfile}
-mmcargs=
-	setenv bootargs console=${console} ${optargs}
-	root=/dev/mmcblk${mmcdev}p${mmcroot} ${raucargs} rootwait rw
-mmcautodetect=yes
-mmcboot=
-	echo Booting from mmc ...;
-	run mmcargs;
-	if test ${no_bootenv} = 0; then
-		if run mmc_load_bootenv; then
-			if test ${dofitboot} = 1; then
-				env import -t ${bootenv_addr_r} ${filesize} overlays;
-			else
-				env import -t ${bootenv_addr_r} ${filesize};
-			fi;
-		fi;
-	fi;
-	run fit_test_and_run_boot;
-	if run loadfdt; then
-		run mmc_apply_overlays;
-		run mmc_apply_extensions;
-		booti ${loadaddr} - ${fdt_addr_r};
-	else
-		echo WARN: Cannot load the DT;
-	fi;
-mmcdev=CONFIG_SYS_MMC_ENV_DEV
-mmcpart=1
-mmcroot=2
-mtdparts=30bb0000.spi:3840k(u-boot),128k(env),128k(env_redund),-(none)
-mtdids=nor0=30bb0000.spi
-spiprobe=true
-netargs=
-	setenv bootargs console=${console} root=/dev/nfs ip=${nfsip} ${optargs}
-	nfsroot=${serverip}:${nfsroot},v3,tcp
-netboot=
-	echo Booting from net ...;
-	if test ${ip_dyn} = yes; then
-		setenv nfsip dhcp;
-		setenv get_cmd dhcp;
-	else
-		setenv nfsip ${ipaddr}:${serverip}::${netmask}::eth0:on;
-		setenv get_cmd tftp;
-	fi;
-	if test ${no_bootenv} = 0; then
-		if run net_load_bootenv; then
-			env import -t ${bootenv_addr_r} ${filesize};
-		fi;
-	fi;
-	${get_cmd} ${loadaddr} ${image};
-	run netargs;
-	if ${get_cmd} ${fdt_addr_r} ${fdtfile}; then
-		run net_apply_overlays;
-		run net_apply_extensions;
-		booti ${loadaddr} - ${fdt_addr_r};
-	else
-		echo WARN: Cannot load the DT;
-	fi;
-nfsroot=/srv/nfs
-no_bootenv=0
-no_extensions=0
-pxefile_addr_r=0x58600000
-ramdisk_addr_r=0x5E000000
-scriptaddr=0x58500000
-script_offset_f=0x400000
-script_size_f=0x2000
-prepare_mcore=setenv optargs "${optargs} clk-imx8mm.mcore_booted"
-update_bootimg=
-	mmc dev ${mmcdev};
-	if dhcp ${loadaddr} ${update_filepath}/${update_filename}; then
-		setexpr fw_sz ${filesize} / 0x200;
-		mmc write ${loadaddr} ${update_offset} ${fw_sz};
-	fi;
-update_filename=flash.bin
-update_offset=0x42
+loadaddr=0x40480000
+fdt_addr=0x48000000
+fdt_high=0xffffffffffffffff
+initrd_high=0xffffffffffffffff
+splashpos=m,m
+splashimage=0x50000000
+
+bootargs=console=ttymxc2,115200 earlycon init=/init firmware_class.path=/vendor/firmware loop.max_part=7 swiotlb=512 fw_devlink.strict=0
+
+bootcmd=
+	echo PHYTEC phyBOARD-Polis Booting with Android 15 Kernel...;
+	if mmc dev 1; then setenv mmcdev 1; else setenv mmcdev 2; fi;
+	mmc read ${loadaddr} 0x8008 0x10000;
+	cp.b ${fdtcontroladdr} ${fdt_addr} 0x20000;
+	booti ${loadaddr} - ${fdt_addr};
diff --git a/configs/phycore-imx8mm_android_defconfig b/configs/phycore-imx8mm_android_defconfig
index 30358ded289..a3a776a273a 100644
--- a/configs/phycore-imx8mm_android_defconfig
+++ b/configs/phycore-imx8mm_android_defconfig
@@ -25,7 +25,7 @@ CONFIG_FIT_EXTERNAL_OFFSET=0x3000
 CONFIG_SPL_LOAD_FIT=y
 CONFIG_OF_SYSTEM_SETUP=y
 CONFIG_USE_BOOTCOMMAND=y
-CONFIG_BOOTCOMMAND="run sr_ir_v2_cmd;run distro_bootcmd;run bsp_bootcmd"
+CONFIG_BOOTCOMMAND="echo PHYTEC phyBOARD-Polis Booting with Android 15 Kernel...; if mmc dev 1; then setenv mmcdev 1; else setenv mmcdev 2; fi; mmc read 0x40480000 0x8008 0x10000; cp.b ${fdtcontroladdr} 0x48000000 0x20000; booti 0x40480000 - 0x48000000"
 CONFIG_BOOTSTD_FULL=y
 CONFIG_DEFAULT_FDT_FILE="imx8mm-phyboard-polis-rdk.dtb"
 CONFIG_SYS_CBSIZE=2048
diff --git a/configs/phycore-imx8mm_android_uuu_defconfig b/configs/phycore-imx8mm_android_uuu_defconfig
index 4d0c1dceb75..154da1829a2 100644
--- a/configs/phycore-imx8mm_android_uuu_defconfig
+++ b/configs/phycore-imx8mm_android_uuu_defconfig
@@ -25,7 +25,7 @@ CONFIG_FIT_EXTERNAL_OFFSET=0x3000
 CONFIG_SPL_LOAD_FIT=y
 CONFIG_OF_SYSTEM_SETUP=y
 CONFIG_USE_BOOTCOMMAND=y
-CONFIG_BOOTCOMMAND="run sr_ir_v2_cmd;run distro_bootcmd;run bsp_bootcmd"
+CONFIG_BOOTCOMMAND="echo PHYTEC phyBOARD-Polis Booting with Android 15 Kernel...; if mmc dev 1; then setenv mmcdev 1; else setenv mmcdev 2; fi; mmc read 0x40480000 0x8008 0x10000; cp.b ${fdtcontroladdr} 0x48000000 0x20000; booti 0x40480000 - 0x48000000"
 CONFIG_BOOTSTD_FULL=y
 CONFIG_DEFAULT_FDT_FILE="imx8mm-phyboard-polis-rdk.dtb"
 CONFIG_SYS_CBSIZE=2048
diff --git a/include/configs/phycore_imx8mm_android.h b/include/configs/phycore_imx8mm_android.h
index bfdd87d0aff..e6dfeda2350 100644
--- a/include/configs/phycore_imx8mm_android.h
+++ b/include/configs/phycore_imx8mm_android.h
@@ -12,6 +12,13 @@
 #undef CFG_EXTRA_ENV_SETTINGS
 #undef CONFIG_BOOTCOMMAND
 
+#define CONFIG_BOOTCOMMAND \
+	"echo PHYTEC phyBOARD-Polis Booting with Android 15 Kernel...; " \
+	"if mmc dev 1; then setenv mmcdev 1; else setenv mmcdev 2; fi; " \
+	"mmc read 0x40480000 0x8008 0x10000; " \
+	"cp.b ${fdtcontroladdr} 0x48000000 0x20000; " \
+	"booti 0x40480000 - 0x48000000"
+
 #define CFG_EXTRA_ENV_SETTINGS		\
 	"splashpos=m,m\0"			\
 	"splashimage=0x50000000\0"		\
@@ -19,6 +26,7 @@
 	"initrd_high=0xffffffffffffffff\0"	\
 	"emmc_dev=2\0"\
 	"sd_dev=1\0" \
+	"bootargs=console=ttymxc2,115200 earlycon init=/init firmware_class.path=/vendor/firmware loop.max_part=7 swiotlb=512 fw_devlink.strict=0\0"
 
 /* Enable mcu firmware flash */
 #ifdef CONFIG_FLASH_MCUFIRMWARE_SUPPORT
-- 
2.34.1

